#!/usr/bin/env python
PACKAGE = "vocus2"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("c_space",    int_t,    0, "c_space", 3,  0, 3)
gen.add("fuse_feature", int_t, 0, "fuse_feature",    1, 0,   2)
gen.add("fuse_conspicuity",    int_t,    0, "fuse_conspicuity",  1, 0, 2)
gen.add("pyr_struct",   int_t,   0, "pyr_struct",  2, 0, 3)
gen.add("start_layer", int_t, 0, "start_layer", 0, 0, 3)
gen.add("stop_layer", int_t, 0, "stop_layer", 4, 0, 4)
gen.add("center_sigma", double_t, 0, "center_sigma", 2.0, 1.0, 5.0)
gen.add("surround_sigma", double_t, 0, "surround_sigma", 6.0, 1.0, 4.0)
gen.add("n_scales", int_t, 0, "n_scales", 2, 1, 5)
gen.add("normalize", bool_t, 0, "normalize", True)

exit(gen.generate(PACKAGE, "vocus2", "vocus_params"))

